### PR TITLE
Wp 127 queue stat

### PR DESCRIPTION
**Pull request by fblackburn1**
Depends-On: https://github.com/wazo-platform/wazo-call-logd-client/pull/5

### CODE REVIEW

File: integration_tests/suite/test_queue_stats.py [hunk 1]
```diff
@@ -323,3 +330,125 @@ def test_list_queue_statistics_multiple_queues_same_datetime(self):
...
+        results = self.call_logd.queue_statistics.get_by_id(
+            queue_id=1,
+            from_='2020-10-06 00:00:00', until='2020-10-07 00:00:00',
+            interval='hour',
+        )
+        assert_that(results, has_entries(total=equal_to(25)))
+
+        assert_that(
+            results['items'],
+            has_item(has_entries({
```
Comment (pc-m) [inline integration_tests/suite/test_queue_stats.py:359]: has_entries can also be used like this to reduce the amount of typing
```python
has_entries(
    from='2020-10-06T04:00:00+00:00',
    until='2020-10-06T05:00:00+00:00'
   ...
)
```
This does not have to be changed, it's just my personal preference that I'm trying to spread ;)
And it does not work if the key contains an `-`

File: wazo_call_logd/plugins/cdr/plugin.py [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
 
-class Plugin(object):
+class Plugin:
     def load(self, dependencies):
         api = dependencies['api']
         config = dependencies['config']
 
         auth_client = AuthClient(**config['auth'])
-        dao = CallLogDAO(new_db_session(config['db_uri']))
+        dao = DAO(new_db_session(config['db_uri'])).call_log
```
Comment (pc-m) [inline wazo_call_logd/plugins/cdr/plugin.py:22]: An instance of `DAO`, which contains an instance of all DAOs is created here and in the support-center plugin.py
dao should be a dependency that is injected in the dependencies

File: wazo_call_logd/plugins/support_center/schema.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+class QueueStatisticsListRequestSchema(Schema):
+    from_ = fields.DateTime(data_key='from', required=True)
+    until = fields.DateTime(missing=None)
+    qos_threshold = fields.Integer()
+    day_start_time = fields.Time(attribute='start_time')
+    day_end_time = fields.Time(attribute='end_time')
+    week_days = fields.List(fields.Integer(), missing=[1, 2, 3, 4, 5, 6, 7])
+
+    @pre_load
+    def convert_week_days_to_list(self, data):
```
Comment (pc-m) [inline wazo_call_logd/plugins/support_center/schema.py:43]: I'd like a `400` if `week_days` = `42`

File: wazo_call_logd/plugins/support_center/schema.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    not_answered = fields.Integer(attribute='timeout', default=0)
+    saturated = fields.Integer(default=0)
+    blocked = fields.Integer(attribute='blocking', default=0)
+    average_waiting_time = fields.Integer(default=0)
+    answered_rate = fields.Float(default=0.0)
+    quality_of_service = fields.Float(attribute='qos', default=0.0)
+
+
+class QueueStatisticsListRequestSchema(Schema):
+    from_ = fields.DateTime(data_key='from', required=True)
```
Comment (pc-m) [inline wazo_call_logd/plugins/support_center/schema.py:35]: I'd like a `400` if `until` > `from_`

File: wazo_call_logd/plugins/support_center/service.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                next_datetime = current + time_delta
+                if next_datetime > until:
+                    next_datetime = until
+        else:
+            yield from_, until
+
+    def get(self, tenant_uuids, queue_id, **kwargs):
+        queue_stats = list()
+
+        interval = kwargs.pop('interval', None)
```
Comment (pc-m) [inline wazo_call_logd/plugins/support_center/service.py:51]: `<nitpicking>`
If you're going to pop `kwargs` with a default value you might as well add if to the function definition.
```python
def get(self, tenant_uuids, queue_id, interval=None, from_=None, until=None, **kwargs):
```
`</nitpicking>`

File: integration_tests/suite/test_queue_stats.py (hunk 1 above)
Comment (bloom1) [reply integration_tests/suite/test_queue_stats.py:359]: I tried it, but `from` is a reserved word :frowning_face: I have to do `**{'from': '...'}` to use it as a keyword

File: wazo_call_logd/plugins/cdr/plugin.py (hunk 2 above)
Comment (fblackburn1) [reply wazo_call_logd/plugins/cdr/plugin.py:22]: :+1: I created a new issue to refactor this part https://wazo-dev.atlassian.net/browse/WAZO-2021

File: wazo_call_logd/plugins/support_center/schema.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    until = fields.DateTime(missing=None)
+    qos_threshold = fields.Integer()
```
Comment (pc-m) [inline wazo_call_logd/plugins/support_center/schema.py:37]: Positive integers please
a qos threshold of `-42` does not make sense

**Review by pc-m — APPROVED** [review]

**Comment by pc-m** [comment]
recheck

**Review by bloom1 — APPROVED** [review]

**Review by pc-m — DISMISSED** [review]

**Review by pc-m — DISMISSED** [review]

**Review by pc-m — DISMISSED** [review]

**Review by pc-m — APPROVED** [review]

**Pull request merged**